packages/features/auth/lib/next-auth-options.ts

```diff
@@ -54,6 +56,35 @@ export const checkIfUserBelongsToActiveTeam = <T extends UserTeams>(user: T) =>
     return metadata.success && metadata.data?.subscriptionId;
   });
 
+const checkIfUserShouldBelongToOrg = async (
+  email: string
+): Promise<[string | undefined, number | undefined]> => {
+  if (!email || !ORGANIZATIONS_AUTOLINK) return [undefined, undefined];
+  const [username, apexDomain] = email.split("@");
+  const existingOrg = await prisma.team.findFirst({
+    where: {
+      AND: [
+        {
+          metadata: {
+            path: ["isOrganizationVerified"],
+            equals: true,
+          },
+        },
+        {
+          metadata: {
+            path: ["orgAutoAcceptEmail"],
+            equals: apexDomain,
+          },
+        },
+      ],
+    },
+    select: {
+      id: true,
+    },
+  });
+  return [username, existingOrg?.id];
```

Comment:
Making sure we return the forced username from email address handle and orgId if exists